tests/TestCase/AppCakeTestCase.php

```diff
@@ -441,6 +443,21 @@ protected function logout()
     }
 
     protected function loginAsSuperadmin()
```

Comment:
Maybe you can combine loginAsSuperadmin with loginAsCustomer (and all the other necessary logins) with a new method:

`protected function login($userId, $email)`

`protected function loginAsSuperadmin()
{
    return $this->login(Configure::read('test.customerId'), Configure::read('test.loginEmailCustomer'));
}`

All the login functionality could be outsourced into a LoginTrait, so that this is not loaded when tests do not need it.